### ISSUE TITLE
Feature Request: Allow esbuild config adjustments

### ISSUE THREAD

**Issue by joshdavenport** [blomqma/next-rest-framework#128]
Am currently using [Lucia](https://lucia-auth.com/getting) and as a result, during the [build](https://github.com/blomqma/next-rest-framework/blob/4ddfb5a1e6752af722c1e6a20a624902e700554d/packages/next-rest-framework/src/cli/utils.ts#L32), esbuild complains about @node-rs/bcrypt and @node-rs/argon2.

![Screenshot 2024-02-19 at 18 03 56](https://github.com/blomqma/next-rest-framework/assets/757828/44f31e7b-63d5-409e-9028-e00585850d3f)

Definitely not a problem inherent to this library, however I believe I could easily resolve by marking them as external in the esbuild build arguments, but there's no way to do so. That's only based on initial research though, not 100% sure what I need to add. For my purposes I'm likely going to work out the needed change and patch but figured maybe this is a scenario somebody else could encounter.

If this was to be a thing, not sure if this would be best placed as an argument to `generate` (something like `--esbuild-config`) or as an argument to `docsApiRoute` etc. Naturally opening this up means the end user could seriously break stuff, but as long as it came with relevant warnings in the docs then it'd allow workarounds when needed.

**Comment by blomqma** [comment]
I believe this could be solved by using the ESBuild [packages](https://esbuild.github.io/api/#packages) option which excludes all dependencies of an ESBuild entry point from the bundle. Those dependencies are also not needed by Next REST Framework as it only analyses it's own modules usages in the entry points when running the CLI commands.

**Comment by blomqma** [comment]
> I believe this could be solved by using the ESBuild [packages](https://esbuild.github.io/api/#packages) option which excludes all dependencies of an ESBuild entry point from the bundle. Those dependencies are also not needed by Next REST Framework as it only analyses it's own modules usages in the entry points when running the CLI commands.

@joshdavenport I have now created an experimental release `v5.1.4-beta.1` that introduces this ESBuild option for a potential fix, please let me know if upgrading to this version addresses your issue.

**Comment by joshdavenport** [comment]
Hey @blomqma, just returned to the project I was working on last month when I reported tihs. Thanks for looking into this in Feb, really appreciated. 

The packages option introduced in that version did work but it's later removed. Evidently some codebases need it (I'm finding I am) and others explicitly don't (as, looking at commits, it was removed due to issues reported by others).

At the moment I'm still relying on patches but it would be amazing if this could be either an option, or done via opening up esbuild config to extensibility.
